modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/YauaaEnrichment.scala

```diff
@@ -91,23 +91,91 @@ final case class YauaaEnrichment(cacheSize: Option[Int]) extends Enrichment {
    * @param userAgent User agent of the event.
    * @return Attributes retrieved thanks to the user agent (if any), as self-describing JSON.
    */
-  def getYauaaContext(userAgent: String): SelfDescribingData[Json] =
-    SelfDescribingData(YauaaEnrichment.outputSchema, parseUserAgent(userAgent).asJson)
+  def getYauaaContext(userAgent: String, headers: List[String]): SelfDescribingData[Json] =
+    SelfDescribingData(YauaaEnrichment.outputSchema, parseUserAgent(userAgent, headers).asJson)
 
   /**
    * Gets the map of attributes retrieved by YAUAA from the user agent.
    * @return Map with all the fields extracted by YAUAA by parsing the user agent.
    *         If the input is null or empty, a map with just the DeviceClass set to Unknown is returned.
    */
-  def parseUserAgent(userAgent: String): Map[String, String] =
+  def parseUserAgent(userAgent: String, headers: List[String]): Map[String, String] =
     userAgent match {
       case null | "" =>
         YauaaEnrichment.DefaultResult
       case _ =>
-        val parsedUA = uaa.parse(userAgent)
+        val headerMap = headers
+          .map(_.split(": ", 2))
+          .collect {
+            case Array(key, value) => key -> value.replaceAll("^\\s+", "")
+          }
+          .toMap ++ Map("User-Agent" -> userAgent)
+        val parsedUA = uaa.parse(headerMap.asJava)
         parsedUA.getAvailableFieldNamesSorted.asScala
           .map(field => decapitalize(field) -> parsedUA.getValue(field))
           .toMap
-          .filter(_._1 != "__SyntaxError__")
+          .filterKeys(validFields)
     }
+
+  /** Yauaa 7.x added many new fields which are not in the 1-0-4 schema */
+  private val validFields = Set(
```

Comment:
I thought this too! If you think more widely, it's a bigger piece of functionality that is currently missing from Iglu libs:

Given some json (the output of yauaa enrichment) and given a schema (the yauaa 1-0-4) schema, cast the json so it is valid against the schema.  In this case, drop fields and keep just the valid fields.

I did not implement it here for a few reasons:
1. Yauaa enrichment would then have a dependency on Iglu client, to get the schema.  It makes the code change quite a lot bigger, because it introduces a `F[_]` and we have to start handling failures to fetch the schema.
2. It is [a temporary problem](https://github.com/snowplow/snowplow-rdb-loader/issues/1065) that we cannot upgrade the yauaa schema to add extra fields.  Once that problem is fixed then we don't need to maintain this list.

It's a balance.  Weighing up 1 vs 2, and considering we have users waiting for this feature, I decided to implement it the easy way.